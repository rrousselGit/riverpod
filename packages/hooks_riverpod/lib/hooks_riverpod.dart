export 'package:flutter_riverpod/flutter_riverpod.dart';

export 'src/consumer.dart'
    show HookConsumer, HookConsumerWidget, StatefulHookConsumerWidget;
