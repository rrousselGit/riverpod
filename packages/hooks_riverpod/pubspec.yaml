name: hooks_riverpod
description: >
  A reactive caching and data-binding framework.
  Riverpod makes working with asynchronous code a breeze.
version: 2.4.10
homepage: https://riverpod.dev
repository: https://github.com/rrousselGit/riverpod
issue_tracker: https://github.com/rrousselGit/riverpod/issues
funding:
  - https://github.com/sponsors/rrousselGit/

environment:
  sdk: ">=2.17.0 <3.0.0"
  flutter: ">=3.0.0"

dependencies:
  collection: ^1.15.0
  flutter:
    sdk: flutter
  flutter_hooks: '>=0.18.0 <0.21.0'
  flutter_riverpod: 2.4.10
  riverpod: 2.5.0
  state_notifier: ">=0.7.2 <2.0.0"

dev_dependencies:
  flutter_test:
    sdk: flutter
  mockito: ^5.0.0
  pub_semver: ^2.1.3
  pubspec_parse: ^1.2.1