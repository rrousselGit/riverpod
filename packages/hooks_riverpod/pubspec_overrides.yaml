# melos_managed_dependency_overrides: flutter_riverpod,riverpod
dependency_overrides:
  flutter_riverpod:
    path: ../flutter_riverpod
  riverpod:
    path: ../riverpod
