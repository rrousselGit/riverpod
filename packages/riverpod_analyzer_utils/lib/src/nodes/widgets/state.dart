part of '../../nodes.dart';

ClassElement? _findStateWidget(ClassElement node) {
  final type = node.supertype?.typeArguments.firstOrNull;
  if (type == null) return null;

  // May be typed as `State<MyWidget>` or `State<StatefulWidget>`.
  // The latter prevents from finding the widget class.
  if (isFromFlutter.isExactlyType(type) ||
      isFromFlutterRiverpod.isExactlyType(type) ||
      isFromRiverpod.isExactlyType(type) ||
      isFromHooksRiverpod.isExactlyType(type)) {
    return null;
  }

  return type.element.cast<ClassElement>();
}

final class StateDeclaration {
  StateDeclaration._({
    required this.widget,
    required this.element,
    required this.node,
  });

  static StateDeclaration? _parse(ClassDeclaration node) {
    final widget = node.declaredElement.let(_findStateWidget);
    final element = node.declaredElement.let(StateDeclarationElement._parse);

    if (element == null) return null;

    return StateDeclaration._(
      widget: widget.let(StatefulWidgetDeclarationElement._parse),
      element: element,
      node: node,
    );
  }

  final ClassDeclaration node;
  final StatefulWidgetDeclarationElement? widget;
  final StateDeclarationElement element;

  WidgetDeclaration? findWidgetAst() {
    final widgetName = widget?.element.name;
    if (widgetName == null) return null;

    final unit = node.thisOrAncestorOfType<CompilationUnit>()!;

    final widgetClass = unit.declarations
        .whereType<ClassDeclaration>()
        .firstWhereOrNull((e) => e.name.lexeme == widgetName);

    return widgetClass?.widget;
  }
}

final class StateDeclarationElement {
  StateDeclarationElement._({
    required this.widget,
    required this.element,
  });

  static final _cache = _Cache<StateDeclarationElement>();

  static StateDeclarationElement? _parse(ClassElement element) {
    return _cache(element, () {
      final widget = _findStateWidget(element);

      return StateDeclarationElement._(
        element: element,
        widget: widget.let(StatefulWidgetDeclarationElement._parse),
      );
    });
  }

  final ClassElement element;
  final StatefulWidgetDeclarationElement? widget;
}
