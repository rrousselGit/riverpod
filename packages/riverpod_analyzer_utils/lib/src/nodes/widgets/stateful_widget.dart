part of '../../nodes.dart';

bool _isCoreType(DartType type) {
  return isFromFlutter.isExactlyType(type) ||
      isFromFlutterRiverpod.isExactlyType(type) ||
      isFromRiverpod.isExactlyType(type) ||
      isFromHooksRiverpod.isExactlyType(type);
}

ClassElement? _findStateFromReturnType(ClassElement node) {
  final type =
      node.methods.firstWhereOrNull((e) => e.name == 'createState')?.returnType;

  if (type == null) return null;

  // May be typed as `MyState createState()` or `State<MyWidget> createState()`.
  // The latter prevents from finding the state class.
  if (_isCoreType(type)) return null;

  return type.element.cast<ClassElement>();
}

ClassElement? _findStateWithMatchingGeneric(ClassElement node) {
  for (final clazz in node
      // ignore: deprecated_member_use, required to support lower versions of analyzer
      .enclosingElement
      .classes) {
    final type = clazz.supertype;
    if (type != null && isState(type) && _findStateWidget(clazz) == node) {
      return clazz;
    }
  }

  return null;
}

ClassElement? _findState(ClassElement node) {
  return _findStateFromReturnType(node) ?? _findStateWithMatchingGeneric(node);
}

final class StatefulWidgetDeclaration extends WidgetDeclaration {
  StatefulWidgetDeclaration({
    required this.node,
    required this.state,
    required this.element,
  });

  static StatefulWidgetDeclaration? _parse(ClassDeclaration node) {
    final stateClass = node.declaredElement.let(_findState);
    final element = node.declaredElement.let(
      StatefulWidgetDeclarationElement._parse,
    );
    if (element == null) return null;

    return StatefulWidgetDeclaration(
      node: node,
      element: element,
      state: stateClass.let(StateDeclarationElement._parse),
    );
  }

  final StateDeclarationElement? state;
  @override
  final StatefulWidgetDeclarationElement element;
  @override
  final ClassDeclaration node;

  StateDeclaration? findStateAst() {
    final stateName = state?.element.name;
    if (stateName == null) return null;

    final unit = node.thisOrAncestorOfType<CompilationUnit>()!;

    final stateClass = unit.declarations
        .whereType<ClassDeclaration>()
        .firstWhereOrNull((e) => e.name.lexeme == stateName);

    return stateClass?.state;
  }
}

final class StatefulWidgetDeclarationElement extends WidgetDeclarationElement {
  StatefulWidgetDeclarationElement({
    required this.element,
    required this.dependencies,
  });

  static final _cache = _Cache<StatefulWidgetDeclarationElement>();

  static StatefulWidgetDeclarationElement? _parse(ClassElement node) {
    return _cache(node, () {
      final dependencies = DependenciesAnnotationElement._of(node);

      return StatefulWidgetDeclarationElement(
        element: node,
        dependencies: dependencies,
      );
    });
  }

  final ClassElement element;
  @override
  final DependenciesAnnotationElement? dependencies;
}
