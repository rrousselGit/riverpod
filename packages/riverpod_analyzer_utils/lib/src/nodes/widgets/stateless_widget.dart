part of '../../nodes.dart';

final class StatelessWidgetDeclaration extends WidgetDeclaration {
  StatelessWidgetDeclaration._({
    required this.element,
    required this.node,
  });

  static StatelessWidgetDeclaration? _parse(ClassDeclaration node) {
    final element = node.declaredElement.let(
      StatelessWidgetDeclarationElement._parse,
    );
    if (element == null) return null;

    return StatelessWidgetDeclaration._(
      element: element,
      node: node,
    );
  }

  @override
  final StatelessWidgetDeclarationElement element;

  @override
  final ClassDeclaration node;
}

final class StatelessWidgetDeclarationElement extends WidgetDeclarationElement {
  StatelessWidgetDeclarationElement._({required this.dependencies});

  static final _cache = _Cache<StatelessWidgetDeclarationElement>();

  static StatelessWidgetDeclarationElement? _parse(ClassElement node) {
    return _cache(node, () {
      final dependencies = DependenciesAnnotationElement._of(node);

      return StatelessWidgetDeclarationElement._(
        dependencies: dependencies,
      );
    });
  }

  @override
  final DependenciesAnnotationElement? dependencies;
}
