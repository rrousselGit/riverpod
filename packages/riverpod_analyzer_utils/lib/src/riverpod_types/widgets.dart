part of '../riverpod_types.dart';

/// [TypeChecker] from `Widget` from Flutter
const widgetType = TypeChecker.fromName('Widget', packageName: 'flutter');

/// [TypeChecker] from `State` from Flutter
const widgetStateType = TypeChecker.fromName('State', packageName: 'flutter');

/// [TypeChecker for `ConsumerWidget``
const statelessWidgetType = TypeChecker.fromName(
  'StatelessWidget',
  packageName: 'flutter',
);

/// [TypeChecker for `ConsumerWidget``
const statefulWidgetType = TypeChecker.fromName(
  'StatefulWidget',
  packageName: 'flutter',
);

/// [TypeChecker for `ConsumerWidget``
const stateType = TypeChecker.fromName(
  'State',
  packageName: 'flutter',
);

/// [TypeChecker for `ConsumerWidget``
const consumerWidgetType = TypeChecker.fromName(
  'ConsumerWidget',
  packageName: 'flutter_riverpod',
);

/// [TypeChecker for `ConsumerStatefulWidget`
const consumerStatefulWidgetType = TypeChecker.fromName(
  'ConsumerStatefulWidget',
  packageName: 'flutter_riverpod',
);

/// [TypeChecker for `ConsumerState`
const consumerStateType = TypeChecker.fromName(
  'ConsumerState',
  packageName: 'flutter_riverpod',
);

/// [TypeChecker for `HookConsumerWidget`
const hookConsumerWidgetType = TypeChecker.fromName(
  'HookConsumerWidget',
  packageName: 'hooks_riverpod',
);

/// [TypeChecker for `StatefulHookConsumerWidget`
const statefulHookConsumerType = TypeChecker.fromName(
  'StatefulHookConsumerWidget',
  packageName: 'hooks_riverpod',
);

/// [TypeChecker for `HookConsumerWidget`
const hookWidgetType = TypeChecker.fromName(
  'HookWidget',
  packageName: 'hooks_riverpod',
);

/// [TypeChecker for `StatefulHookConsumerWidget`
const statefulHookType = TypeChecker.fromName(
  'StatefulHookWidget',
  packageName: 'hooks_riverpod',
);

bool isState(DartType type) {
  return stateType.isExactlyType(type) || consumerStateType.isExactlyType(type);
}

bool isStatelessWidget(DartType type) {
  return statelessWidgetType.isExactlyType(type) ||
      consumerWidgetType.isExactlyType(type) ||
      hookConsumerWidgetType.isExactlyType(type) ||
      hookWidgetType.isExactlyType(type);
}

bool isStatefulWidget(DartType type) {
  return statefulWidgetType.isExactlyType(type) ||
      consumerStatefulWidgetType.isExactlyType(type) ||
      statefulHookConsumerType.isExactlyType(type) ||
      statefulHookType.isExactlyType(type);
}
