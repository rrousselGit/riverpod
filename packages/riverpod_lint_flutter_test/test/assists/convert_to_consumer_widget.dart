import 'package:flutter/widgets.dart';
import 'package:hooks_riverpod/hooks_riverpod.dart';
import 'package:flutter_hooks/flutter_hooks.dart';

class Stateless extends StatelessWidget {
  const Stateless({super.key});

  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class StatelessWithComma extends StatelessWidget {
  const StatelessWithComma({super.key});

  @override
  Widget build(
    BuildContext context,
  ) {
    return const Placeholder();
  }
}

class Hook extends HookWidget {
  const Hook({super.key});

  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class HookConsumer extends HookConsumerWidget {
  const HookConsumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
  ) {
    return const Placeholder();
  }
}

class Stateful extends StatefulWidget {
  const Stateful({super.key});

  @override
  State<Stateful> createState() => _StatefulState();
}

class _StatefulState extends State<Stateful> {
  /// Hello world
  @override
  Widget build(BuildContext context) {
    // Some comments
    return const Placeholder();
  }
}

class ExplicitCreateState extends StatefulWidget {
  const ExplicitCreateState({super.key});

  @override
  ExplicitCreateStateState createState() => ExplicitCreateStateState();
}

class ExplicitCreateStateState extends State<ExplicitCreateState> {
  @override
  Widget build(
    BuildContext context,
  ) {
    return const Placeholder();
  }
}

class HookStateful extends StatefulHookWidget {
  const HookStateful({super.key});

  @override
  State<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends State<HookStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class ConsumerStateful extends ConsumerStatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class HookConsumerStateful extends StatefulHookConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }
}

class Consumer extends ConsumerWidget {
  const Consumer({super.key});

  @override
  Widget build(
    BuildContext context,
    WidgetRef ref,
  ) {
    return const Placeholder();
  }
}
