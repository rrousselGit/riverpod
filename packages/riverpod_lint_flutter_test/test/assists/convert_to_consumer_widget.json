[{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":266,"length":0,"replacement":", WidgetRef ref"},{"offset":168,"length":15,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":454,"length":0,"replacement":", WidgetRef ref"},{"offset":342,"length":15,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":608,"length":0,"replacement":", WidgetRef ref"},{"offset":520,"length":10,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":678,"length":18,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":993,"length":174,"replacement":""},{"offset":990,"length":0,"replacement":"/// Hello world\n  @override\n  Widget build(BuildContext context, WidgetRef ref) {\n    // Some comments\n    return const Placeholder();\n  }\n"},{"offset":927,"length":62,"replacement":""},{"offset":876,"length":14,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":31,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":1350,"length":165,"replacement":""},{"offset":1347,"length":0,"replacement":"@override\n  Widget build(\n    BuildContext context, WidgetRef ref,\n  ) {\n    return const Placeholder();\n  }\n"},{"offset":1265,"length":81,"replacement":""},{"offset":1203,"length":14,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":1676,"length":142,"replacement":""},{"offset":1673,"length":0,"replacement":"@override\n  Widget build(BuildContext context, WidgetRef ref) {\n    return const Placeholder();\n  }\n"},{"offset":1603,"length":69,"replacement":""},{"offset":1544,"length":18,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":2008,"length":159,"replacement":""},{"offset":2005,"length":0,"replacement":"@override\n  Widget build(BuildContext context, WidgetRef ref) {\n    return const Placeholder();\n  }\n"},{"offset":1918,"length":86,"replacement":""},{"offset":1851,"length":22,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}},{"priority":35,"change":{"message":"Convert to ConsumerWidget","edits":[{"fileStamp":0,"edits":[{"offset":2383,"length":167,"replacement":""},{"offset":2380,"length":0,"replacement":"@override\n  Widget build(BuildContext context, WidgetRef ref) {\n    return const Placeholder();\n  }\n"},{"offset":2279,"length":100,"replacement":""},{"offset":2204,"length":26,"replacement":"ConsumerWidget"}]}],"linkedEditGroups":[]}}]