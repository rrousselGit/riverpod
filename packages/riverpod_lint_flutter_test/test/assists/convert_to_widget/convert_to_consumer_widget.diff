Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:5`:
```
import 'package:flutter_hooks/flutter_hooks.dart';

- class Stateless extends StatelessWidget {
-   const Stateless({super.key});
- 
-   @override
-   Widget build(BuildContext context) {
+ class Stateless extends ConsumerWidget {
+   const Stateless({super.key});
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:14`:
```
}

- class StatelessWithComma extends StatelessWidget {
-   const StatelessWithComma({super.key});
- 
-   @override
-   Widget build(
-     BuildContext context,
+ class StatelessWithComma extends ConsumerWidget {
+   const StatelessWithComma({super.key});
+ 
+   @override
+   Widget build(
+     BuildContext context, WidgetRef ref,
  ) {
    return const Placeholder();
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:25`:
```
}

- class Hook extends HookWidget {
-   const Hook({super.key});
- 
-   @override
-   Widget build(BuildContext context) {
+ class Hook extends ConsumerWidget {
+   const Hook({super.key});
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:34`:
```
}

- class HookConsumer extends HookConsumerWidget {
+ class HookConsumer extends ConsumerWidget {
  const HookConsumer({super.key});

```
---
Message: `Convert to ConsumerWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:46`:
```
}

- class Stateful extends StatefulWidget {
-   const Stateful({super.key});
- 
-   @override
-   State<Stateful> createState() => _StatefulState();
- }
- 
- class _StatefulState extends State<Stateful> {
-   /// Hello world
-   @override
-   Widget build(BuildContext context) {
+ class Stateful extends ConsumerWidget {
+   const Stateful({super.key});
+ 
+  
+   /// Hello world
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    // Some comments
    return const Placeholder();
```
---
Message: `Convert to ConsumerWidget`
Priority: 31
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:62`:
```
}

- class ExplicitCreateState extends StatefulWidget {
-   const ExplicitCreateState({super.key});
- 
-   @override
-   ExplicitCreateStateState createState() => ExplicitCreateStateState();
- }
- 
- class ExplicitCreateStateState extends State<ExplicitCreateState> {
-   @override
-   Widget build(
-     BuildContext context,
-   ) {
+ class ExplicitCreateState extends ConsumerWidget {
+   const ExplicitCreateState({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:78`:
```
}

- class HookStateful extends StatefulHookWidget {
-   const HookStateful({super.key});
- 
-   @override
-   State<HookStateful> createState() => HookStatefulState();
- }
- 
- class HookStatefulState extends State<HookStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class HookStateful extends ConsumerWidget {
+   const HookStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:92`:
```
}

- class ConsumerStateful extends ConsumerStatefulWidget {
-   const ConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
- }
- 
- class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class ConsumerStateful extends ConsumerWidget {
+   const ConsumerStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:106`:
```
}

- class HookConsumerStateful extends StatefulHookConsumerWidget {
-   const HookConsumerStateful({super.key});
- 
-   @override
-   ConsumerState<HookConsumerStateful> createState() =>
-       _HookConsumerStatefulState();
- }
- 
- class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
-   @override
-   Widget build(BuildContext context) {
+ class HookConsumerStateful extends ConsumerWidget {
+   const HookConsumerStateful({super.key});
+ 
+  
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:133`:
```
}

- class StatelessWithField extends StatelessWidget {
-   const StatelessWithField({
-     super.key,
-     required this.field,
-   });
- 
-   final int field;
-   static final int staticField = 42;
- 
-   @override
-   Widget build(BuildContext context) {
+ class StatelessWithField extends ConsumerWidget {
+   const StatelessWithField({
+     super.key,
+     required this.field,
+   });
+ 
+   final int field;
+   static final int staticField = 42;
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
    return Column(
      children: [
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:153`:
```
}

- class HookConsumerWithField extends HookConsumerWidget {
+ class HookConsumerWithField extends ConsumerWidget {
  const HookConsumerWithField({
    super.key,
```
---
Message: `Convert to ConsumerWidget`
Priority: 35
Diff for file `test/assists/convert_to_widget/convert_to_widget.dart:180`:
```
}

- class ConsumerStatefulWithField extends ConsumerStatefulWidget {
-   const ConsumerStatefulWithField({
-     super.key,
-     required this.field,
-     required this.foo,
-   });
- 
-   final int field;
-   final FooClass foo;
-   static final int staticField = _constantNumber;
- 
-   @override
-   ConsumerState<ConsumerStatefulWithField> createState() =>
-       _ConsumerStatefulWithFieldState();
- }
- 
- const _constantNumber = 42;
- 
- class _ConsumerStatefulWithFieldState
-     extends ConsumerState<ConsumerStatefulWithField> {
-   void printFoo() {
-     print(widget.foo);
-   }
- 
-   @override
-   Widget build(BuildContext context) {
-     print(widget.field);
-     printFoo();
-     return Column(
-       children: [
-         Text('${widget.field}'),
-         Text('${widget.foo.bar}'),
-         Text('${ConsumerStatefulWithField.staticField}'),
-       ],
-     );
-   }
- }
+ class ConsumerStatefulWithField extends ConsumerWidget {
+   const ConsumerStatefulWithField({
+     super.key,
+     required this.field,
+     required this.foo,
+   });
+ 
+   final int field;
+   final FooClass foo;
+   static final int staticField = _constantNumber;
+ 
+  
+   void printFoo() {
+     print(foo);
+   }
+ 
+   @override
+   Widget build(BuildContext context, WidgetRef ref) {
+     print(field);
+     printFoo();
+     return Column(
+       children: [
+         Text('$field'),
+         Text('${foo.bar}'),
+         Text('$staticField'),
+       ],
+     );
+   }
+ }
+ 
+ const _constantNumber = 42;

```
---
