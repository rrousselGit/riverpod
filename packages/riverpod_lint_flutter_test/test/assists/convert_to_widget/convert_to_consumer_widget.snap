Message: `Convert to ConsumerWidget`
<<<< start: 168 -- end: 265
StatelessWidget {
  const Stateless({super.key});

  @override
  Widget build(BuildContext contex
==== start: 168 -- end: 279
ConsumerWidget {
  const Stateless({super.key});

  @override
  Widget build(BuildContext context, WidgetRef re
>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 342 -- end: 453
StatelessWidget {
  const StatelessWithComma({super.key});

  @override
  Widget build(
    BuildContext contex
==== start: 342 -- end: 467
ConsumerWidget {
  const StatelessWithComma({super.key});

  @override
  Widget build(
    BuildContext context, WidgetRef re
>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 520 -- end: 607
HookWidget {
  const Hook({super.key});

  @override
  Widget build(BuildContext contex
==== start: 520 -- end: 626
ConsumerWidget {
  const Hook({super.key});

  @override
  Widget build(BuildContext context, WidgetRef re
>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 678 -- end: 682
Hook
==== start: 678 -- end: 678
>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 876 -- end: 1166
StatefulWidget {
  const Stateful({super.key});

  @override
  State<Stateful> createState() => _StatefulState();
}

class _StatefulState extends State<Stateful> {
  /// Hello world
  @override
  Widget build(BuildContext context) {
    // Some comments
    return const Placeholder();
  }

==== start: 876 -- end: 1069
ConsumerWidget {
  const Stateful({super.key});

  
/// Hello world
  @override
  Widget build(BuildContext context, WidgetRef ref) {
    // Some comments
    return const Placeholder();
  }
}

>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 1203 -- end: 1514
StatefulWidget {
  const ExplicitCreateState({super.key});

  @override
  ExplicitCreateStateState createState() => ExplicitCreateStateState();
}

class ExplicitCreateStateState extends State<ExplicitCreateState> {
  @override
  Widget build(
    BuildContext context,
  ) {
    return const Placeholder();
  }

==== start: 1203 -- end: 1377
ConsumerWidget {
  const ExplicitCreateState({super.key});

  
@override
  Widget build(
    BuildContext context, WidgetRef ref,
  ) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 1544 -- end: 1817
StatefulHookWidget {
  const HookStateful({super.key});

  @override
  State<HookStateful> createState() => HookStatefulState();
}

class HookStatefulState extends State<HookStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1544 -- end: 1702
ConsumerWidget {
  const HookStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 1859 -- end: 2166
StatefulWidget {
  const ConsumerStateful({super.key});

  @override
  ConsumerState<ConsumerStateful> createState() => _ConsumerStatefulState();
}

class _ConsumerStatefulState extends ConsumerState<ConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 1859 -- end: 2013
Widget {
  const ConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
Message: `Convert to ConsumerWidget`
<<<< start: 2204 -- end: 2549
StatefulHookConsumerWidget {
  const HookConsumerStateful({super.key});

  @override
  ConsumerState<HookConsumerStateful> createState() =>
      _HookConsumerStatefulState();
}

class _HookConsumerStatefulState extends ConsumerState<HookConsumerStateful> {
  @override
  Widget build(BuildContext context) {
    return const Placeholder();
  }

==== start: 2204 -- end: 2370
ConsumerWidget {
  const HookConsumerStateful({super.key});

  
@override
  Widget build(BuildContext context, WidgetRef ref) {
    return const Placeholder();
  }
}

>>>>
